Simplified/NYPLUserAccount.swift

```diff
@@ -0,0 +1,402 @@
+import Foundation
+
+extension Notification.Name {
+  static let NYPLUserAccountDidChange = Notification.Name("NYPLUserAccountDidChangeNotification")
+  static let NYPLUserAccountLoginDidChange = Notification.Name("NYPLUserAccountLoginDidChangeNotification")
+}
+
+@objc extension NSNotification {
+  public static let NYPLUserAccountDidChange = Notification.Name.NYPLUserAccountDidChange
+  public static let NYPLUserAccountLoginDidChange = Notification.Name.NYPLUserAccountLoginDidChange
+}
+
+@objcMembers class NYPLUserAccount : NSObject {
+  static private let shared = NYPLUserAccount()
+  static private let accountInfoLock = NSRecursiveLock()
```

Comment:
did you make this static to protect access to keychain because `NYPLKeychain` is not thread safe? routing all NYPLUserAccount instances through this shared lock makes me think of potential deadlocks. It would be safer to have it as a normal instance var unless I'm missing something. 

If `NYPLKeychain` is not thread-safe it'd be better to have that class handle its own threadsafety internally instead of fixing it here. This is of course out of scope for this work.